extensions/homeassistant/src/components/fan.tsx

```diff
@@ -0,0 +1,72 @@
+import { ActionPanel, Color, Icon, Action, Keyboard } from "@raycast/api";
+import { ha } from "../common";
+import { State } from "../haapi";
+import { ensureMinMax } from "../utils";
+
+function getSpeedValues(step: number): number[] {
```

Comment:
Most of this code is adapted from `light`, so thanks for that as a starting point!